backend/app/api/upload.py

```diff
@@ -40,6 +45,51 @@ def post(self):
                 image.save(path, image.format)
             #filename = photos.save(request.files['photos'])
             return "Success", 201
+        elif 'interactions' in request.files:
+            print "In interactions!!"
+            print request.form
+            interaction_id = request.form.get('interaction_id')
+            dir_path = os.path.join(dest, interactions_sub, interaction_id)
+            print "Path is: " + dir_path
+            if not os.path.exists(dir_path):
+                print "making the dir!"
+                os.makedirs(dir_path)
+            path = os.path.join(dir_path, request.files['interactions'].filename)
+            if os.path.exists(path):
+                os.remove(path)
+            print request.files
+            filename = interactions_upload_set.save(request.files.get('interactions'), interaction_id)
+            print filename
+            return "Success", 201
         else:
             msg = 'No file included in upload request'
             abort(400, message=msg)
+        
+class InteractionFiles(Resource):
+    def get(self, id):
+        filenames = self.getFileList(id);
+        response = {"filenames": filenames}
+        return response, 200
+    
+    # This will delete any files that are not in the posted list
+    def post(self, id):
+        path = os.path.join(dest, interactions_sub, id)
+        existing_files = self.getFileList(id)
+        posted_files = list(request.get_json())
+        files_to_delete = [x for x in existing_files if x not in posted_files]
```

Comment:
Could this be done in a generator comprehension since it is just used to loop over below? Generators are faster and take less memory.